apollo-router/src/plugins/expose_query_plan.rs

```diff
@@ -237,7 +305,11 @@ mod tests {
         )
         .await;
 
-        insta::assert_json_snapshot!(serde_json::to_value(response).unwrap());
+        // Since we're exposing the query plan, the extensions better not be empty!
+        assert!(!response.extensions.is_empty());
+
+        // Since this is a dry-run, we shouldn't have any data
+        assert!(response.data.is_none());
```

Comment:
@abernix: here's how we testing this now (as opposed to how it was being tested before with snapshots, which would require updating the snapshots each time the query plan changed); when we do a dry-run, we want there to be something in the extensions and yet nothing in the data; when we're doing the real thing, we want both the extensions and data; when we aren't using the plugin, we don't want the extension to be populated (the base case, showing that what's in the extension is all and only the query plan)